doc/reliability.md

```diff
@@ -0,0 +1,27 @@
+
+## BroadcastSessionsManager
+
+Orchestrators are managed by a `BroadcastSessionsManager` stored in the `rtmpConnections` on the `LivepeerServer` interface. The sessions manager is initiated when the RTMP stream is registered by `gotRTMPStreamHandler`. The orchestrator list is first populated then, when `refreshSessions` is called within `NewSessionManager`.
+
+The `BroadcastSessionsManager` stores orchestrators in two lists, a `sessList` and a `sessMap`.  The `sessList` is an array containing a "working" list of orchestrators. When a Broadcaster is in need of an orchestrator, it selects and removes it from `sessList` only if said Orchestrator also exists in `sessMap`. Therefore, `sessMap` contains all orchestrators currently in use or available for use. It is a map with a string key of the URI of the orchestrator, and a value of that orchestrator's `BroadcastSession`.
+
+## Orchestrator List Refresh
+
+The orchestrator list is refreshed when the number of sessions in `sessList` is less than double the `HTTMPTimeout` in seconds (hard-coded to 8 seconds at the moment) divided by the lenght of segments (hard-coded to 2 seconds at the moment) OR less than the size of the OrchestratorPool saved on disk, whichever is less (i.e. when its length is less than what is required to keep in memory). This happens at startup (as described above), and when an orchestrator is selected for individual transcoding in `selectSession`.
+
+## Orchestrator Selection
+
+To give preference to O's that respond with transcoded segments quickly, instead of selecting an Orchestrator from the beginning of `sessList` when needed, and placing new Orchestrators that are finished processing a segment at the end, `selectSession` takes Orchestrators from the end of `sessList`. If transcoding is successful, it adds them back to the end of `sessList`. 
+
+## Transcoding Errors & Retries
+
+If there is an error uploading segment to an Orchestrator's OS, submitting the segment to an Orchestrator, downloading transcoded segments, or the segment signature check fails, the Orchestrator is removed from the `sessMap`. The segment is retried with a different Orchestrator. When `selectSession` is called in this retry scenario, though the removed session might still exist in `sessList`, only a session that still exists in `sessMap` will be selected.  If there is no error in segment transcoding, `completeSession` adds session back to `sessList`.
```

Comment:
I think it will be good to add that retries stops when `sessMap` is empty.

Reply:
I'll add that!